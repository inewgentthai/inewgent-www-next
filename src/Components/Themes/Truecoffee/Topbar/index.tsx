// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { ITopbar } from "./Topbar";
import Link from "next/link";
// #endregion Local Imports

const Topbar: React.FunctionComponent<ITopbar.IProps> = (): JSX.Element => {
    return (
      <section className="topbar fixed-top">
        <div className="container">
          <div className="row no-gutters">
            <div className="col text-left">
              <div className="icn-history float-left">
                <Link href={`/order`} as={`/order`}>
                  <a href="order-history.html" data-toggle="tooltip" data-placement="right" title="Order History" />
                </Link>
              </div>
            </div>
            <div className="col-6 text-center">
              <Link href={`/`} as={`/`}>
                <a>
                  <img src="static/truecoffee-ordering/images/logo.svg" />
                </a>
              </Link>
            </div>
            <div className="col text-right">
              <div className="icn-cart float-right">
                <Link href={`/cart`} as={`/cart`}>
                  <a href="cart.html"><div className="icn-noti">1</div></a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </section>
    );
};

export { Topbar };
