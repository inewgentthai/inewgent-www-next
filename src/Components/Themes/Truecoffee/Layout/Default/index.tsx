// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { IDefaultLayout } from "./DefaultLayout";
import { CartBar } from "../../CartBar";
import { CategoryBar } from "../../CategoryBar";
import { Topbar } from "../../Topbar";
import { Modal } from "../../Modal";
import { Header } from "../../Header";
import { Footer } from "../../Footer";
import { Loading } from "@Components/Loading";
// #endregion Local Imports

const DefaultLayout: React.FunctionComponent<IDefaultLayout.IProps> = (
  { children, title }
  ): JSX.Element => {
    return (
     <div>
        <Header title={ title }/>
        <Loading/>
        <div className="wrapper">
          <Topbar />
          <CategoryBar />
          { children }
          <CartBar />
          <Modal />
        </div>
        <Footer />
      </div>
    );
};

export { DefaultLayout };
