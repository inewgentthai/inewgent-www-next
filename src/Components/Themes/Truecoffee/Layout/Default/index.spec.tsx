// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { render } from "@Test/utils";
import { DefaultLayout } from "@Components/Themes/Truecoffee/Layout";
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
// #endregion Local Imports

describe("DefaultLayout", () => {
    const initialState = {
        menu: {
            data: [],
            content: [],
            loading: true,
            modal: false
        },
        loading: {
            status: true
        }
    }
    const mockStore = configureStore()
    let store

    it("should match snapshot", () => {
        store = mockStore(initialState)
        const { container } = render(
        <Provider store={store}>
            <DefaultLayout title='Test'>Test</DefaultLayout>
        </Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
