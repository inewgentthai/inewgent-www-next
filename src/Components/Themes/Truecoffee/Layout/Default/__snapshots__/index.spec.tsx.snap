// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout should match snapshot 1`] = `
<div>
  <div>
    <section
      id="loading"
    >
      <div
        class="loading-icon"
      />
    </section>
    <div
      class="wrapper"
    >
      <section
        class="topbar fixed-top"
      >
        <div
          class="container"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col text-left"
            >
              <div
                class="icn-history float-left"
              >
                <a
                  data-placement="right"
                  data-toggle="tooltip"
                  href="order-history.html"
                  title="Order History"
                />
              </div>
            </div>
            <div
              class="col-6 text-center"
            >
              <a
                href="/"
              >
                <img
                  src="static/truecoffee-ordering/images/logo.svg"
                />
              </a>
            </div>
            <div
              class="col text-right"
            >
              <div
                class="icn-cart float-right"
              >
                <a
                  href="cart.html"
                >
                  <div
                    class="icn-noti"
                  >
                    1
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="categories"
      >
        <div
          class="container"
        >
          <div
            class="categories-list row justify-content-center"
          >
            <div
              class="item capsule"
            >
              <a
                href="#"
              >
                Allsss
              </a>
            </div>
            <div
              class="item capsule active"
            >
              <a
                href="#"
              >
                Coffee
              </a>
            </div>
            <div
              class="item capsule"
            >
              <a
                href="#"
              >
                Tea
              </a>
            </div>
            <div
              class="item capsule"
            >
              <a
                href="#"
              >
                Non Coffee
              </a>
            </div>
            <div
              class="item capsule"
            >
              <a
                href="#"
              >
                Tea
              </a>
            </div>
            <div
              class="item capsule"
            >
              <a
                href="#"
              >
                Non Coffee
              </a>
            </div>
            <div
              class="item capsule"
            >
              <a
                href="#"
              >
                Tea
              </a>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col cat-found text-center"
            >
              <b
                class="text-red"
              >
                “Coffee”
              </b>
               Found 8 item
            </div>
          </div>
        </div>
      </section>
      Test
      <section
        class="cart-bar fixed-bottom"
      >
        <div
          class="container"
        >
          <a
            href="cart.html"
          >
            <div
              class="row no-gutters justify-content-between align-items-center"
            >
              <div
                class="col-4"
              >
                <div
                  class="item-in-cart badge"
                >
                  3
                </div>
              </div>
              <div
                class="col go-to-cart"
              >
                Go to Cart
              </div>
              <div
                class="col-4 total"
              >
                <div>
                  Total
                </div>
                <div>
                  ฿9,999.00
                </div>
              </div>
            </div>
          </a>
        </div>
      </section>
      <div />
    </div>
    <button
      id="fabbtt"
      title="Go to top"
    >
      Back to Top
    </button>
  </div>
</div>
`;
